src/context/analysisResultStateSlice.ts

```diff
@@ -386,6 +403,65 @@ export const requestAndStoreAnalysis = createAsyncThunk<
   );
 });
 
+export const requestAndStorePolygonAnalysis = createAsyncThunk<
+  PolygonAnalysisResult,
+  PolygonAnalysisDispatchParams,
+  CreateAsyncThunkTypes
+>('analysisResultState/requestAndStorePolygonAnalysis', async (params, api) => {
+  const { adminLevel, hazardLayer, startDate, endDate, extent } = params;
+
+  const zones = await fetchAdminLayerGeoJSON(adminLevel);
+
+  const classes = await fetchWMSLayerAsGeoJSON({
+    lyr: hazardLayer,
+    startDate,
+    endDate,
+  });
+
+  const result = calculate({
+    zones,
+    zone_properties: [getAdminNameProperty(adminLevel)],
+    classes,
+    class_properties: hazardLayer?.zonal?.class_properties, // eslint-disable-line camelcase
```

Comment:
What happens if there is no class_properties?

Reply:
if no class_properties are provided, zonal will give an id to each hazard feature in the order that they appear in the hazardLayer and use that as the class property.